### ISSUE TITLE
Please consider dual licence

### ISSUE THREAD

**Issue by AndrewD** [jddiener/I2C-eTPU#1]
Would you consider licensing this project with a second, more common, licence: eg BSD or Apache.

I would like to avoid the complication of getting a new licence approved for our use for just one library.

We actively contribute to several projects, this request is purely to minimise non-technical overhead.

This appears to be a nice clean library and I'd prefer to use it instead of creating another solution.

**Comment by AndrewD** [comment]
I just noticed most of your other repo are using the MIT license which would also be ok for our usage.

**Comment by jddiener** [comment]
I dual-licensed it.  If you end up having feedback on the software, it is appreciated.

-john

**Comment by AndrewD** [comment]
Thanks for doing this, it's greatly appreciated. I'll let you know how we go. 

One observation was that it requires 4 TPU channels; I can understand this requirement with some early eTPU implementations like 5554, but I suspect it's feasible to do bidirectional data on a single channel with more recent devices. Is this something you have looked into?

**Comment by jddiener** [comment]
Agree, it would be interesting to generate a 2-channel version.  This was discussed some 2-3 years ago but didn’t happen.  The original versions of this code were written quite a long time ago, back when MPC55xx support was more important.  It’s been quite a while since I’ve gotten into the details of the code, but there is a good chance it is a relatively straight-forward enhancement.  I ran a couple of the sim tests and didn’t spot any resource usage issues that would for sure force larger design changes.  But it was a cursory look.

-john

**Comment by AndrewD** [comment]
> Agree, it would be interesting to generate a 2-channel version.  This was discussed some 2-3 years ago but didn’t happen.

 > ... there is a good chance it is a relatively straight-forward enhancement.  I ran a couple of the sim tests and didn’t spot any resource usage issues that would for sure force larger design changes.  But it was a cursory look.

It would be interesting, but after some successful testing patching to old hardware we have committed 4 pins to this.

So far the microcode is working nicely: I didn't use the provided cpu layer but we have had no issues so far.

I think I have one or two minor microcode changes, mainly to allow for a smarter driver with better error management when queuing multiple transactions. I can provide a patch if you are interested.

My only other thought is that it would be interesting to have a scaled down build option to potentially reduce tpu load, but that is not a concern at the moment.
